File: python/Prodi.py
from Course import Course
from Dosen import Dosen
from Mahasiswa import Mahasiswa

class Prodi: #prodi has a course
    
    #constructor
    def __init__(self, nama_prodi = "", kode = "", course = Course(), dosen = Dosen(), mahasiswa = Mahasiswa()):
        self.nama_prodi = nama_prodi
        self.kode = kode
        self.course = course
        self.dosen = dosen
        self.mahasiswa = mahasiswa
        
    #setter prodi, kode, dan course
    def setNama_prodi(self, nama_prodi):
        self.nama_prodi = nama_prodi
        
    def setKode(self, kode):
        self.kode = kode
    
    def setCourse(self, course):
        self.course = course
    
    def setCourse(self, course):
        self.course = course
    
    def setDosen(self, dosen):
        self.dosen = dosen
        
    def setMahasiswa(self, mahasiswa):
        self.mahasiswa = mahasiswa
    
    #getter prodi, kode, dan course
    def getNama_prodi(self):
        return self.nama_prodi
    
    def getKode(self):
        return self.kode
    
    def getCourse(self):
        return self.course
    
    def getDosen(self):
        return self.dosen
    
    def getMahasiswa(self):
        return self.mahasiswa

File: c++/Program/Prodi.cpp
#include"Header.hh"

Prodi::Prodi(){
    nama_prodi = "";
    kode = "";
}

Prodi::Prodi(string nama_prodi, string kode, Course course, Mahasiswa mhs, Dosen dosen){
    this->nama_prodi = nama_prodi;
    this->kode = kode;
    this->course = course;
    this->mhs = mhs;
    this->dosen = dosen;
}

void Prodi::setNamaProdi(string nama_prodi){
    this->nama_prodi = nama_prodi;
}
string Prodi::getNamaProdi(){
    return this->nama_prodi;
}

void Prodi::setKode(string kode){
    this->kode = kode;
}
string Prodi::getKode(){
    return this->kode;
}

void Prodi::setCourse(Course course){
    this->course = course;
}
Course Prodi::getCourse(){
    return this->course;
}

void Prodi::setMhs(Mahasiswa mhs){
    this->mhs = mhs;
}
Mahasiswa Prodi::getMhs(){
    return this->mhs;
}

void Prodi::setDosen(Dosen dosen){
    this->dosen = dosen;
}
Dosen Prodi::getDosen(){
    return this->dosen;
}

Prodi::~Prodi(){
    
}